### PR TITLE
shaka-packager: 3.2.0 -> 3.3.0

### PR DESCRIPTION
**Pull request by niklaskorz**
Changelogs:

- https://github.com/shaka-project/shaka-packager/releases/tag/v3.2.1
- https://github.com/shaka-project/shaka-packager/releases/tag/v3.3.0

I left both updates as separate commits in case anyone needs v3.2.1 specifically (so they can look into the history and see what's required to build that). Overriding just the version and commit hash would not be sufficient to get v3.2.1 after this PR, as I removed a patch that was required before in the v3.3.0 update commit.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/sh/shaka-packager/0002-Unvendor-dependencies.patch [hunk 1]
```diff
@@ -1,12 +1,13 @@
-From 4c7743f4fd231648d7bd1513c9db71a651ef7553 Mon Sep 17 00:00:00 2001
+From a412a1cbce2b8f28e831aabe0c8d31832eb3f840 Mon Sep 17 00:00:00 2001
 From: Niklas Korz <niklas@niklaskorz.de>
-Date: Wed, 4 Sep 2024 14:52:23 +0200
+Date: Thu, 24 Oct 2024 22:05:50 +0200
 Subject: [PATCH 2/3] Unvendor dependencies
```
[the remark this answers is not in the document]
Comment (niklaskorz) [reply pkgs/by-name/sh/shaka-packager/0002-Unvendor-dependencies.patch]: Removed the patch headers entirely so this isn't part of the diffs in future updates

**Review by GetPsyched — APPROVED** [review]
Diff LGTM

**Comment by GaetanLepage** [comment]
nit: can you get rid of the `with lib;` in the `meta` attrs ?

**Comment by niklaskorz** [comment]
> nit: can you get rid of the `with lib;` in the `meta` attrs ?

I was trying to find any docs that clear up why this should be avoided, as there are over 20 thousand packages that follow the same pattern, even some by you. Care to elaborate and maybe we can add that elaboration to nixpkgs docs? Also that should be a treewide cleanup if it becomes recommended practice.

**Comment by GaetanLepage** [comment]
> I was trying to find any docs that clear up why this should be avoided, as there are over 20 thousand packages that follow the same pattern, even some by you. Care to elaborate and maybe we can add that elaboration to nixpkgs docs? Also that should be a treewide cleanup if it becomes recommended practice.

I could not find an exact statement that officially discourages it.
I think that people tend to remove any occurrence of the `with XXX;` pattern as in certain cases it can behave counter-intuitively.

I have been removing those as I come across them. I am not aware of any tree-wide initiative to do so.
Feel free to ignore this suggestion ofc.

**Comment by GetPsyched** [comment]
> Care to elaborate and maybe we can add that elaboration to nixpkgs docs?

While not in the nixpkgs manual, it's documented [here](https://nix.dev/guides/best-practices#with-scopes).

There is also #208242 as a tracking issue for these uses (while primarily focusing on NixOS modules -- along with [several PRs from Stunkymonkey to fix the same](https://github.com/NixOS/nixpkgs/pulls?q=is%3Apr+%22treewide%22+%22with+lib%22+author%3AStunkymonkey)). There are some comments discussing `with lib;` specifically for `meta`, but I couldn't find those.

> Also that should be a treewide cleanup if it becomes recommended practice.

^this. Updating the expression would be cool, but out of scope of this PR.

**Comment by niklaskorz** [comment]
> > Care to elaborate and maybe we can add that elaboration to nixpkgs docs?
> 
> While not in the nixpkgs manual, it's documented [here](https://nix.dev/guides/best-practices#with-scopes).
> 
> There is also #208242 as a tracking issue for these uses (while primarily focusing on NixOS modules -- along with [several PRs from Stunkymonkey to fix the same](https://github.com/NixOS/nixpkgs/pulls?q=is%3Apr+%22treewide%22+%22with+lib%22+author%3AStunkymonkey)). There are some comments discussing `with lib;` specifically for `meta`, but I couldn't find those.
> 
> > Also that should be a treewide cleanup if it becomes recommended practice.
> 
> ^this. Updating the expression would be cool, but out of scope of this PR.

Thanks for pointing out the issue! I'm not opposed to including the change in this PR (and also committed it a moment ago), but just wanted to point out it is a very common practice in nixpkgs and will continue to be one as long as the majority of packages you look at for reference make use of it.

**Comment by GaetanLepage** [comment]
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 351014`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>shaka-packager</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>shaka-packager</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>shaka-packager</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>shaka-packager</li>
  </ul>
</details>

**Review by GaetanLepage — APPROVED** [review]
LGTM !

**Review by GetPsyched — APPROVED** [review]

**Pull request merged**